spec/sidekiq_adhoc_job/utils/class_inspector_spec.rb

```diff
@@ -41,4 +41,23 @@
       end
     end
   end
+
+  context 'with an inherited class without redefining method' do
+    before do
+      stub_const("TempWorker", Class.new(SidekiqAdhocJob::Test::DummyWorker))
+    end
+
+    let(:klass) { TempWorker }
```

Comment:
Perhaps this suffices.

```suggestion
    let(:klass) { Class.new(SidekiqAdhocJob::Test::DummyWorker) }
```

Reply:
@gohkhoonhiang Thank you for the suggestion. I did try it in the first place, it worked for this spec, but broke many other specs when running the whole suite `bundle exec rspec` with error like this:
```
     NoMethodError:
       undefined method 'start_with?' for nil
     # ./lib/sidekiq_adhoc_job/strategy.rb:33:in 'block in SidekiqAdhocJob::Strategy#allowed_namespace?'
     # ./lib/sidekiq_adhoc_job/strategy.rb:33:in 'Array#any?'
     # ./lib/sidekiq_adhoc_job/strategy.rb:33:in 'SidekiqAdhocJob::Strategy#allowed_namespace?'
     # ./lib/sidekiq_adhoc_job/strategy.rb:24:in 'block in SidekiqAdhocJob::Strategy#load'
     # ./lib/sidekiq_adhoc_job/strategy.rb:21:in 'ObjectSpace.each_object'
     # ./lib/sidekiq_adhoc_job/strategy.rb:21:in 'Enumerator#each'
     # ./lib/sidekiq_adhoc_job/strategy.rb:21:in 'SidekiqAdhocJob::Strategy#load'
     # ./lib/sidekiq_adhoc_job/worker_classes_loader.rb:7:in 'SidekiqAdhocJob::WorkerClassesLoader.load'
     # ./lib/sidekiq_adhoc_job.rb:32:in 'SidekiqAdhocJob.init'
     # ./spec/support/sidekiq_adhoc_job_setup.rb:22:in 'block (2 levels) in <top (required)>'
```

Originated from this line: https://github.com/gohkhoonhiang/sidekiq_adhoc_job/blob/fb1ada720f1cb66be4d359b87d0ade3c4020eb99/lib/sidekiq_adhoc_job/strategy.rb#L21

The reason is that `Class.new(SidekiqAdhocJob::Test::DummyWorker)` object remains in `ObjectSpace` even after the spec ended. Using `stub_const` is safer and can avoid that issue.